### PR TITLE
Removed IRelease Interface

### PR DESCRIPTION
**Pull request by apankratovantonp**

### CODE REVIEW

File: inference-engine/include/details/ie_so_pointer.hpp [hunk 1]
```diff
@@ -208,17 +152,4 @@ class SOPointer {
...
- * @brief Creates a special shared_pointer wrapper for the given type from a specific shared module
- * @tparam T An type of object SOPointer can hold
- * @param name Name of the shared library file
- * @return A created object
- */
-template <class T>
-inline std::shared_ptr<T> make_so_pointer(const std::string & name) = delete;
-
-template <class T>
-inline std::shared_ptr<T> make_so_pointer(const std::wstring & name) = delete;
```
Comment (ilya-lavrenov) [inline inference-engine/include/details/ie_so_pointer.hpp:222]: this function is used in samples. Removing this right now breaks BW compatibility. 
Can we keep it for some time, but deprecate maybe?

File: inference-engine/include/ie_extension.h [hunk 2]
```diff
@@ -102,27 +99,7 @@ class INFERENCE_ENGINE_API_CLASS(Extension) : public IExtension {
     /**
      * @brief A SOPointer instance to the loaded templated object
      */
+    details::SharedObjectLoader::Ptr so;
     details::SOPointer<IExtension> actual;
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_extension.h:103]: I suppose `details::SOPointer<IExtension> actual;` already contains link to `details::SharedObjectLoader::Ptr`, is not it?

File: inference-engine/include/details/ie_so_pointer.hpp (hunk 1 above)
Comment (apankratovantonp) [reply inference-engine/include/details/ie_so_pointer.hpp:222]: This function is used just for extension so leaved just extension version and added deprecation warning

File: inference-engine/include/ie_extension.h (hunk 2 above)
Comment (apankratovantonp) [reply inference-engine/include/ie_extension.h:100]: removed

**Pull request closed without merging**